addons/binding/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/protocol/RokuCommunication.java

```diff
@@ -0,0 +1,209 @@
+     * map.put(app, id);
+     * }
+     * return map;
+     * }
+     */
+
+    public void updateState(RokuState state) throws IOException {
+        Document doc = getRequest(ROKU_DEVICE_INFO);
+        String[] methodStringArray = { "udn", "serial-number", "device-id", "vendor-name", "model-name", "model-number",
+                "model-region", "wifi-mac", "ethernet-mac", "network-type", "user-device-name", "software-version",
+                "software-build", "power-mode", "headphones-connected" };
+        doc.getDocumentElement().normalize();
+        if (doc != null) {
+            NodeList nList = doc.getElementsByTagName("device-info");
+            for (int i = 0; i < nList.getLength(); i++) {
+                Node nNode = nList.item(i);
+                logger.debug("Current Element: " + nNode.getNodeName());
+                if (nNode.getNodeType() == Node.ELEMENT_NODE) {
+                    for (int ii = 0; ii < methodStringArray.length; ii++) {
+                        Element eElement = (Element) nNode;
+                        Class<RokuState> aClass = RokuState.class;
+                        Field field = null;
+                        try {
+                            field = aClass.getField(methodStringArray[ii].replace("-", "_"));
+                            field.set(state, getTagName(methodStringArray[ii], eElement));
+                        } catch (NoSuchFieldException | SecurityException | IllegalArgumentException
+                                | IllegalAccessException e) {
+                            logger.error("Method not found {}", methodStringArray[ii].replace("-", "_"));
+                        }
+                    }
+                }
+            }
+        }
+        doc = getRequest(ROKU_ACTIVE_APP);
+        doc.getDocumentElement().normalize();
+        if (doc != null) {
+            NodeList nList = doc.getElementsByTagName("active-app");
+            for (int i = 0; i < nList.getLength(); i++) {
+                Node nNode = nList.item(i);
+                Element eElement = (Element) nNode;
+                logger.debug("Current Element: " + nNode.getNodeName());
+                try {
+                    state.active_app = getTagName("screensaver", eElement);
+                    String app_value = getSubTagName("screensaver", eElement);
+                    try {
+                        state.active_app_img = new RawType(getData("/query/icon/" + app_value));
+                    } catch (Exception e) {
+                        logger.debug("Failed to get channel artwork for: {}", e);
+                    }
+                } catch (NullPointerException e) {
+                    state.active_app = getTagName("app", eElement);
+                    String app_value = getSubTagName("app", eElement);
+                    try {
+                        state.active_app_img = new RawType(getData("/query/icon/" + app_value));
+                    } catch (Exception e1) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception